### ISSUE TITLE
自己创建的parameter和函数为什么不优化？需要加什么操作吗？？比如这个例子

### ISSUE THREAD

**Issue by jianjianGJ** [PaddlePaddle/Paddle#21295]
```import paddle.fluid as fluid
import paddle.fluid.layers as layers
place = fluid.CPUPlace()
exe = fluid.Executor(place)
W = layers.create_parameter(shape=[1], dtype='float32')
y = 2*W;
adam = fluid.optimizer.Adam(learning_rate=0.1)
adam.minimize(y,parameter_list=[W])
exe.run(fluid.default_startup_program())
for i in range(3):
    result,w = exe.run(fluid.default_main_program(),
                         fetch_list=[y, W])
    print("y:",result,";w:",w)

```
输出：
```
y: [2.383992] ;w: [1.191996]
y: [2.383992] ;w: [1.191996]
y: [2.383992] ;w: [1.191996]
```

**Comment by dongfangyixi** [comment]
use W.name instead of W when passing parameter list may help:    
```python
adam.minimize(y,parameter_list=[W.name])
```

**Comment by GT-ZhangAcer** [comment]
可以试试通过喂数据的方法，传递一个确切的数据喂给该网络

**Comment by jianjianGJ** [comment]
我用了`paddle.fluid.backward.append_backward(loss, parameter_list=None, no_grad_set=None, callbacks=None)`这个API后可以了
